lib/storage/src/content_manager/toc/collection_meta_ops.rs

```diff
@@ -315,6 +316,28 @@ impl TableOfContent {
                 )?;
                 collection.finish_shard_transfer(transfer).await?;
             }
+            ShardTransferOperations::SnapshotRecovered(transfer) => {
+                // Validate transfer exists to prevent double handling
+                shard_transfer::validate_transfer_exists(
+                    &transfer,
+                    &collection.state().await.transfers,
+                )?;
+
+                // Set shard state from `PartialSnapshot` to `Partial`
+                let operation = SetShardReplicaState {
+                    collection_name: collection_id,
+                    shard_id: transfer.shard_id,
+                    peer_id: transfer.to,
+                    state: ReplicaState::Partial,
+                    from_state: Some(ReplicaState::PartialSnapshot),
+                };
+                log::debug!(
+                    "Set shard replica state from {:?} to {:?} after snapshot recovery",
+                    ReplicaState::PartialSnapshot,
+                    ReplicaState::Partial,
+                );
+                self.set_shard_replica_state(operation).await?;
+            }
```

Comment:
Implementation of consensus operation:

- assert a transfer is still active
- set shard replica state from `PartialSnapshot` to `Partial`

Reply:
Just to be clear, it checks the transfer is still active only because it is going to finalize it, right?

Reply:
Not because it will finalize it, but because it will go into the next stage of the transfer. We want an explicit check for the transfer to be active to prevent a rogue transfer from breaking the cluster state.